src/sparkpilot/config.py

```
@@ -96,6 +99,21 @@ def _validate_environment_mode(settings: Settings) -> bool:
         raise ValueError(
             "SPARKPILOT_DRY_RUN_MODE=true is only allowed in development/test environments."
         )
+    if not is_dev_like_environment:
+        if settings.database_url == _DEFAULT_DATABASE_URL:
+            raise ValueError(
+                "SPARKPILOT_DATABASE_URL must be explicitly set in non-development environments. "
+                "The default localhost URL must not be used in staging or production."
+            )
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)